src/api/java/com/minecolonies/api/colony/buildings/PickUpPriorityState.java

```diff
@@ -0,0 +1,69 @@
+package com.minecolonies.api.colony.buildings;
+
+public enum PickUpPriorityState
+{
+    NEVER(0),
+    STATIC(1),
+    AUTOMATIC(2);
+
+    /**
+     * The internal serializable int-representation of this enum.
+     * We could use Java's ordinal(), but that is VERY error-prone.
+     * So, do our own instead.
+     */
+    private final int intRepresentation;
+
+    PickUpPriorityState(final int intRepresentation)
+    {
+        this.intRepresentation = intRepresentation;
+    }
+
+    /**
+     * The integer representation for this enum. Used for message serialization.
+     *
+     * @return The integer representation
+     */
+    public int getIntRepresentation()
+    {
+        return intRepresentation;
+    }
+
+    /**
+     * Creates the enum from an int. Used for message deserialization.
+     * This is not DRY, but we're talking about 3 to maybe later 4 values here.
+     *
+     * @param intRepresentation The internal representation of the enum.
+     * @return The corresponding enum.
+     */
+    public static PickUpPriorityState fromIntRepresentation(int intRepresentation)
```

Comment:
Why do we need the int representation, can't we pick the ordinal?

Reply:
Because enum ordinals in Java are not meant to be used in that way. I mentioned it in line 10 of the enum.

There's a couple of reasons for this, see https://stackoverflow.com/a/44654410/886407.

Reply:
Because Java's ordinals are not meant to be used for such scenarios.
There's a couple of reasons for this, see https://stackoverflow.com/a/44654410.

Reply:
Yeah, it definitely would lead to backwards compatibility problems if you insert values inbetween. Due to laziness all of our enums use the ordenal for persistency. 

I wonder if it is possible to abstract this a bit so that it could be used anywhere basically (so that we can avoid adding this boilerplate to all our enums).

Reply:
Backwards compatibility isn't even my main concern. [This is.](https://www.tutorialspoint.com/java/lang/enum_ordinal.htm)
A pointer to the enum will return 0, not the value you would expect. This will undoubtedly cause a bug along the way.

As for boilerplate, there might be a way by using enum types inheritance and/or generic shenanigans. However, I feel this is slightly out of the scope of this PR^^

Reply:
Sounds okay, although, whoever will create an enum next, will have to extract this to avoid copying the code just over.